src/platform/macos/view.rs

```diff
@@ -351,8 +351,27 @@ extern fn key_down(this: &Object, _sel: Sel, event: id) {
         let state = &mut *(state_ptr as *mut ViewState);
         let window_id = WindowId(get_window_id(state.window));
 
+        state.raw_characters = {
+            let characters: id = msg_send![event, characters];
+            let slice = slice::from_raw_parts(
+                characters.UTF8String() as *const c_uchar,
+                characters.len(),
+            );
+            let string = str::from_utf8_unchecked(slice);
+            Some(string.to_owned())
+        };
+
         let keycode: c_ushort = msg_send![event, keyCode];
-        let virtual_keycode = to_virtual_key_code(keycode);
+        let virtual_keycode = {
+            let mut v_code = to_virtual_key_code(keycode);
+            if v_code.is_none() {
```

Comment:
It's cleaner to use [`or_else`](https://doc.rust-lang.org/core/option/enum.Option.html#method.or_else).